src/query/storage/m3/storagemetadata/attributes.go

```diff
@@ -64,3 +64,26 @@ func (a Attributes) String() string {
 		a.Retention.String(),
 		a.Resolution.String())
 }
+
+func (a Attributes) CombinedWith(other Attributes) Attributes {
+	metricType := a.MetricsType
+	if other.MetricsType == AggregatedMetricsType {
+		metricType = other.MetricsType
```

Comment:
nit: `metricType = AggregatedMetricsType` just to not think that it's actually will be set to AggMeType every time if is true?